### ISSUE TITLE
fetch CGPM Resolution 1889-00 has stopped working

### ISSUE THREAD

**Issue by opoudjis** [relaton/relaton-bipm#41]
````
bundle exec relaton fetch "BIPM CGPM Resolution 1889-00"
Calling `DidYouMean::SPELL_CHECKERS.merge!(error_name => spell_checker)' has been deprecated. Please call `DidYouMean.correct_error(error_name, spell_checker)' instead.
[relaton] (BIPM CGPM Resolution 1889-00) not found.
````

**Comment by andrew2net** [comment]
@opoudjis because the identifiers were updated https://github.com/relaton/relaton-bipm/commits/main#issuecomment-1292238497
To fetch the document we can use `CGPM -- Resolution (1889)`, `CGPM -- Résolution (1889)`, `CGPM -- RES (1889, EN)`, or `CGPM -- RES (1889, EN)` references.
Should we support previous references?

**Comment by opoudjis** [comment]
Well... they're still what is documented in the README. It's a question for @ronaldtse, but I suspect the answer is no.

**Comment by ronaldtse** [comment]
Yeah let’s not support old syntax.

**Comment by opoudjis** [comment]
Ok

**Comment by ronaldtse** [comment]
> The decision to use "--" is disruptive to me, because Asciidoctor automatically converts "--" to em-dashes (and of course, in real world typology, as opposed to ASCII workarounds, that's what the symbol actually is.) I can change Metanorma to universally convert " — " back to " -- ", but I need a firm commitment from you that "—" will never show up in a Relaton identifier.

I don't know if that's possible, because Relaton titles should be as what titles given are. The real issue is that text values in Relaton (title, abstract, names, etc) never had a "textual schema" defined. This is something that we need to work on.

**Comment by opoudjis** [comment]
This is not about titls

> > The decision to use "--" is disruptive to me, because Asciidoctor automatically converts "--" to em-dashes (and of course, in real world typology, as opposed to ASCII workarounds, that's what the symbol actually is.) I can change Metanorma to universally convert " — " back to " -- ", but I need a firm commitment from you that "—" will never show up in a Relaton identifier.
> 
> I don't know if that's possible, because Relaton titles should be as what titles given are. The real issue is that text values in Relaton (title, abstract, names, etc) never had a "textual schema" defined. This is something that we need to work on.

This is not about titles, this is about identifiers, which should be more tightly constrained to begin with, since they are used to invoke external lookups. 

The problem is that "--" is a default Asciidoctor substitution (and it is reasonable for it to be a default Asciidoctor substitution). And indeed, "---" should never have been used within an identifier. It is a glaring misfeature that we are now stuck with.

I've implemented a workaround, but surprises are unwelcome to me, and this was a surprise.

**Comment by opoudjis** [comment]
And relaton fetch "BIPM DECN CIPM/101-1 (2012, EN)" DOES NOT EVEN WORK.

This is not acceptable, @andrew2net. Major changes to functionality like this need to be documented, and I need to be warned of them.

**Comment by opoudjis** [comment]
bundle exec relaton fetch "CCTF -- Recommendation 1 (1970)" works
bundle exec relaton fetch "CCTF -- REC 1 (1970)" does not

So there seems to be a systematic issue with short codes...

**Comment by andrew2net** [comment]
> ... well, this is a very confusing and unwelcome inconsistency, and it needs to be VERY clearly documented in the readme. People have to be able to generate these references.

Documented it here https://github.com/relaton/relaton-bipm#reference-structures-for-cctf-ccds-cgpm-cipm-documents

> And relaton fetch "BIPM DECN CIPM/101-1 (2012, EN)" DOES NOT EVEN WORK.

Fixed in relaton v 1.14.1

> As of this writing: I still need:
> 
> BIPM DECN CIPM/105-1 (2016, EN) BIPM CGPM -- Resolution (1889)
> 
> to work, or alternatives if those aren't right.

Fixed in relaton v 1.14.1

**Comment by andrew2net** [comment]
> bundle exec relaton fetch "CCTF -- Recommendation 1 (1970)" works bundle exec relaton fetch "CCTF -- REC 1 (1970)" does not
> 
> So there seems to be a systematic issue with short codes...

Fixed in relaton-bipm v 1.14.1

**Comment by opoudjis** [comment]
I ended up changing the references I used, because I couldn't get the ones I provided to work. If there are ongoing problems, they can be raised by the editors, not me.
